Add version to footer

little9 opened the pull request:
This commit adds a version to the footer. This includes the
branch or tag that the software is on along with the SHA. The
SHA is only visible if you hover over the branch name.

This is based on the instructions at:
https://curationexperts.github.io/playbook/every_project/git_sha.html

It is using this path to `revisions.log`:

`/opt/alexandria/revisions.log`

little9 commented:
![screenshot-2018-2-26 results for alexandria digital research library](https://user-images.githubusercontent.com/4324761/36702066-3a519406-1b1b-11e8-9a83-db38e45f26dc.png)

little9 commented:
See curationexperts/alexandria#43

dunn commented on config/initializers/git_sha.rb:
```diff
@@ -0,0 +1,26 @@
+revisions_logfile = "/opt/alexandria/revisions.log"
```
This path should probably be defined in `config/application.yml` since that's where we specify most (ideally all) of our server-specific paths.

mark-dce replied:
If you haven't modified the default capistrano directory structure (http://capistranorb.com/documentation/getting-started/structure/) the `revisions.log` file will always be two directories above the application root, so you don't really need to explicitly define a path for this, you could instead infer it:
```
revisions_logfile = File.join( File.expand_path('..//..', Rails.root), 'revisions.log' )
```
There's probably even something more elegant than that, but the main point is we can infer the location instead of having to explicitly set it, which might make setup simpler.

dunn commented on app/views/shared/_footer.html.erb:
```diff
@@ -3,6 +3,9 @@
       Copyright 2014–<%= Date.today.year %> The Regents of the University of California, All Rights Reserved.
     </div>
+    <div class="version">
+      Version <span title="<%= GIT_SHA %>"><%= BRANCH %> updated <%= LAST_DEPLOYED %></span>
```
I don't mind having the SHA in parentheses after the version string.  Or maybe I'm just worried that using `title` like this isn't properly semantic.

little9 replied:
I think it might look too weird to have the SHA down there. Do you mind if we just get rid of it?

little9 commented:
@dunn I changed this so that the SHA isn't in the title attribute and changed it so that it references the revisions.log file in the way that @mark-dce suggests above

dunn commented:
@chrissyrissmeyer I'm happy with this, but I shouldn't be the only one to sign off on UI changes.

chrissyrissmeyer commented:
That is fine with me too! Great idea.
